independent-projects/arc/runtime/src/main/java/io/quarkus/arc/RequestContext.java

```diff
@@ -104,6 +110,11 @@ public void activate(Collection<ContextInstanceHandle<?>> initialState) {
             }
         }
         currentContext.set(state);
+        // Fire an event with qualifier @Initialized(RequestScoped.class)
+        Set<Annotation> qualifiers = new HashSet<>(4);
+        qualifiers.add(Initialized.Literal.REQUEST);
+        qualifiers.add(Any.Literal.INSTANCE);
+        EventImpl.createNotifier(Object.class, Object.class, qualifiers, ArcContainerImpl.instance()).notify(toString());
```

Comment:
I think we should cache non-empty notifiers and skip notifications for empty notifiers.